### ISSUE TITLE
Compiler warnings result in "Build failed"

### ISSUE THREAD

**Issue by ebkalderon** [AtomBuild/atom-build-cargo#10]
I often work with unstable features in Nightly Rust. Currently, I have the "Automatically build on save" option turned on. Whenever I save a file, the build panel at the bottom reports in red "Build failed", even though my projects were built successfully.

Could the error matching be tuned to recognize compiler warnings as a successful build?

**Comment by ebkalderon** [comment]
Okay, changing:

```
'(?<file>.+.rs):(?<line>\\d+):(?<col>\\d+):',
```

to this:

```
'(?<file>.+.rs):(?<line>\\d+):(?<col>\\d+):(error):',
```

Seems to have fixed it. Should I add this to the pull request?

**Comment by colin-kiegel** [comment]
Personally I like the current _match everything_ style, but I guess its a matter of taste and you could argue that matching warnings is a bit redundant with linter.

Is there a general preferred behaviour across atom-build targets for this question? I think most importantly this should be consistent for the while atom-build family, as long as there is no setting for it.

**Comment by ebkalderon** [comment]
Currently, it is only a suggestion (noseglid/atom-build#225). If solely for the sake of consistency with other build targets, warnings should indeed result in a build failure.

Not that I'm advocating this as acceptable behavior from an IDE, to be clear. I applaud good programming practices and the elimination of all warnings, but sometimes chasing down every last warning is simply not possible given time constraints and other limitations. As such, the build status should not result in failure if the compiler produces a valid compiled and linked executable.

That's my view of it, anyway. I encourage further discussion and more community input on the matter.

**Comment by colin-kiegel** [comment]
PS: Maybe we _can_ do both
- exclude matching warnings from debug builds  (maybe bench too)
- include matching warnings with release and test builds

You could pick between the extremes 'builds with warnings' and 'passes all tests without warnings'. What do you think about this compromise?

**Comment by ebkalderon** [comment]
Sounds good to me! Shall I create a pull request?

**Comment by noseglid** [comment]
Fixed with #12
